File: typescript/src/problems101_200/single-number.ts
class Problem {
  singleNumber(nums: number[]): number {
    throw new Error(`Not implemented (${nums}).`);
  }
}

class Default extends Problem {
  singleNumber(nums: number[]): number {
    const result: number[] = [];
    for (const num of nums) {
      const index: number = result.indexOf(num);
      if (index === -1) {
        result.push(num);
      } else {
        result.splice(index, 1);
      }
    }
    return result[0];
  }
}

const SOLUTIONS = [new Default()];
export default SOLUTIONS;

File: python/src/problems101_200/single_number.py
from abc import ABC, abstractmethod
from typing import override


class Problem(ABC):
    @abstractmethod
    def single_number(self, nums: list[int]) -> int:
        pass


class Default(Problem):
    @override
    def single_number(self, nums: list[int]) -> int:
        result: list[int] = []
        for num in nums:
            if num not in result:
                result.append(num)
            else:
                result.remove(num)
        return result[0]


SOLUTIONS: list[Problem] = [Default()]
